tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Microsoft.Crank.Agent/ProcessUtil.cs

```
@@ -136,34 +138,74 @@ public static async Task<ProcessResult> RunAsync(
+                    {
+                        sys_kill(process.Id, sig: 2); // SIGINT
 
-                    var cancel = new CancellationTokenSource();
+                        var cancel = new CancellationTokenSource();
 
-                    await Task.WhenAny(processLifetimeTask.Task, Task.Delay(TimeSpan.FromSeconds(5), cancel.Token));
+                        await Task.WhenAny(processLifetimeTask.Task, Task.Delay(TimeSpan.FromSeconds(5), cancel.Token));
 
-                    cancel.Cancel();
+                        cancel.Cancel();
+                    }
+
+                    if (!process.HasExited)
+                    {
+                        process.CloseMainWindow();
+
+                        if (!process.HasExited)
+                        {
+                            process.Kill();
+                        }
+                    }
+                    break;
                 }
 
-                if (!process.HasExited)
+                if (trackStatistics)
                 {
-                    process.CloseMainWindow();
-
-                    if (!process.HasExited)
+                    try
                     {
-                        process.Kill();
+                        // CPU usage
+                        TimeSpan currentCpuTime = process.TotalProcessorTime;
+                        double cpuUsage = (currentCpuTime - lastCpuTime).TotalMilliseconds / (1000.0 * Environment.ProcessorCount) * 100;
+                        lastCpuTime = currentCpuTime;
+                        cpuSamples.Add(cpuUsage);
+
+                        // Memory usage
+                        process.Refresh(); // Refresh to get updated memory info
+                        long memoryUsage = process.WorkingSet64;
+                        memorySamples.Add(memoryUsage);
+                    }
+                    catch (InvalidOperationException)
+                    {
+                        // Process may have exited while checking stats, ignore
```

Let's at least log a warning if we swallow the exception.